components/movementsensor/utils.go

```diff
@@ -84,12 +84,38 @@ type LastError struct {
 	count int     // How many items in errs are non-nil
 }
 
+// LastPosition stores the last position seen by the movement sensor.
+type LastPosition struct {
+	lastposition *geo.Point
+	mu           sync.Mutex
+}
+
 // NewLastError creates a LastError object which will let you retrieve the most recent error if at
 // least `threshold` of the most recent `size` items put into it are non-nil.
 func NewLastError(size, threshold int) LastError {
 	return LastError{errs: make([]error, size), threshold: threshold}
 }
 
+// NewLastPosition creates a new point that's (NaN, NaN)
+// go-staticcheck.
+func NewLastPosition() LastPosition {
```

Comment:
I think you could use this to replas everywhere you're setting `geo.Point(math.Nan, math.Nan)

Reply:
That can potentially overwrite a valid point that we are using in other place. I think it's safer to just use it to initialize LastPostion.